src/api/config.c

```diff
@@ -767,6 +767,20 @@ static int api_config_patch(struct ftl_conn *api)
 
```

Comment:
In the line above we have `config_changed=true;`
\> should this be placed after the new value validation? We reject the new config when the value is invalid in the end.

Otherwise manual editing of `pihole.toml` can still add invalid entries.

Reply:
It doesn't matter. If validation fails, this function will free the temporary `newconf` in line 776 below and return. Nobody will ever look at the value of `config_changed`